### PR TITLE
[Todoist] View menubar task in raycast command

### LINKED ISSUE #7057: [Todoist] Extension crashes after clicking 'Home' command
### Extension

https://www.raycast.com/doist/todoist

### Description

After installing extension and logging into TODOist, I'm unable to view any of my task. The reason seems to be pretty obvious. 

#### Error:
```
TypeError: Cannot read properties of null (reading 'due')

l7:TodayTasks.tsx:16:51
Ir:index.js:6:5510
    at ray-list
ut:index.js:20:23404
```

### Steps To Reproduce

1. Install extension
2. Log into Todoist via browser
3. Get back to Raycast
4. Open Raycast -> Home (Todoist extension command)
5. Error

### Current Behaviour

_No response_

### Expected Behaviour

_No response_

### PR DESCRIPTION
**Pull request by casassg**
## Description

Make viewing tasks on raycast possible. This is useful as when you are working on a task you want to see details without necessarily launching the whole app. Specially since raycast will just be launched on top of your current window vs todoist launching the app. Makes for fast manner of checking details within the same raycast system. 

Some other things included:
- Add subtask count to TaskDetail component 
- Fix #7057 -> issue where user does not edit any detail from task causing to cache a null 

## Screencast

Screenshot 2023-06-09 at 11 43 40 AM
Screenshot 2023-06-09 at 11 43 34 AM

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

File: extensions/todoist/src/components/MenubarTask.tsx [hunk 1]
```diff
@@ -99,6 +99,14 @@ const MenuBarTask = ({ task, data, setData }: MenuBarTaskProps) => {
   return (
     <View>
       <MenuBarExtra.Submenu title={removeMarkdown(task.content)} icon={getPriorityIcon(task)}>
+        <MenuBarExtra.Item
```
Comment (thomaslombart) [inline extensions/todoist/src/components/MenubarTask.tsx:102]: What do you think of naming it `Open in Raycast` with the `Icon.RaycastLogoNeg` icon to make it clearer?

**Review by thomaslombart — COMMENTED** [review]
Nice addition!

File: extensions/todoist/src/components/MenubarTask.tsx (hunk 1 above)
Comment (casassg) [reply extensions/todoist/src/components/MenubarTask.tsx:102]: sgtm! changing

**Comment by casassg** [comment]
@thomaslombart any way to check if cachedData is corrupted easily and if so force a refresh? Would love to add it here, but haven't found a way to check for errors within `useSyncData` and no way to force refresh either.

File: extensions/todoist/src/components/TaskDetail.tsx [hunk 2]
```diff
@@ -72,6 +73,14 @@ export default function TaskDetail({ taskId }: TaskDetailProps): JSX.Element {
 
                 <Detail.Metadata.Label title="Due Date" text={displayedDate} icon={Icon.Calendar} />
 
+                {subTasks && subTasks?.length > 0 ? (
+                  <Detail.Metadata.Label
+                    title={"Sub-tasks"}
+                    text={`${subTasks.length} subtask` + (subTasks.length > 1 ? "s" : "")}
```
Comment (sebiq96) [inline extensions/todoist/src/components/TaskDetail.tsx:79]: What do you think about: ``${subTasks.length} subtask${subTasks.length > 1 ? "s" : ""}``

Comment (casassg) [reply extensions/todoist/src/components/TaskDetail.tsx:79]: im pretty new to JS in general so never sure whats best practice. lt me modify and push

**Comment by casassg** [comment]
@thomaslombart any chance to merge this?

File: extensions/todoist/package.json [hunk 3]
```diff
@@ -169,12 +169,12 @@
...
         {
-          "name": "focusFeatureWidth",
+          "name": "taskWidth",
           "type": "textfield",
           "required": false,
           "default": "40",
-          "title": "Maximum Width of Focused Task",
-          "description": "Maximum width of the focused task's text."
+          "title": "Maximum Width of Menu-bar Task",
+          "description": "Maximum width of all menu-bar task's text."
```
[the remark this answers is not in the document]
Comment (casassg) [reply extensions/todoist/package.json]: it happened if you follow https://github.com/raycast/extensions/issues/7057 repro steps. When editing a task but not changing anything, todoist api returns an empty array via sync api as no changes happened. This issue is that then a task w value nil was set in the cachedData which broke it all.

**Review by thomaslombart — APPROVED** [review]

**Pull request merged**